sim/priest/power_infusion.go

```diff
@@ -0,0 +1,75 @@
+package priest
+
+import (
+	"github.com/wowsims/tbc/sim/core"
+	"github.com/wowsims/tbc/sim/core/stats"
+)
+
+var PowerInfusionCooldownID = core.NewCooldownID()
+
+func (priest *Priest) registerPowerInfusionCD() {
+
+	actionID := core.ActionID{SpellID: 10060, CooldownID: PowerInfusionCooldownID, Tag: int32(priest.RaidIndex)}
+	
+	baseManaCost := priest.BaseMana() * 0.16
+
+	powerInfusionCD := core.PowerInfusionCD
+	
+	remainingPowerInfusionUsages := 0
+	
+	var powerInfusionTarget *core.Character
+	
+	priest.AddMajorCooldown(core.MajorCooldown{
+		ActionID:   actionID,
+		CooldownID: PowerInfusionCooldownID,
+		Cooldown:   powerInfusionCD,
+		UsesGCD:    true,
+		Type:       core.CooldownTypeMana,
+		CanActivate: func(sim *core.Simulation, character *core.Character) bool {
+			// Does there need to be logic around this?
```

Comment:
CanActivate is for checking conditions that should completely prevent casting. In this case, you need to check that the character has enough mana. Should be able to find examples in other files.

ShouldActivate is for conditions that are not absolutely necessary but are good for ideal usage. ShouldActivate gets ignored when the user sets a cooldown timing in the UI. For PI I don't think there are any such conditions.

Reply:
My PI priest in classic would disagree w/ checking if the target has enough mana... but what do we even arbitrarily consider enough?